snow/validators/retriever_test.go

```diff
@@ -0,0 +1,92 @@
+package validators
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/flare-foundation/flare/ids"
+	"github.com/flare-foundation/flare/utils/constants"
+)
+
+func TestCachingRetriever_GetValidators(t *testing.T) {
+	testBlockID := ids.ID{
+		0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+		0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F,
+		0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17,
+		0x18, 0x19, 0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F,
+	}
+
+	t.Run("cache call count", func(t *testing.T) {
```

Comment:
I'm not sure what you are testing for here. You are creating some kind of internal logic of the mock that tests how often the mock uses the cache. Rather than that, you can simply count how often the underlying retriever is being called, and every time it is not being called, the validators are served from the cache. You don't need to simulate the internal logic for this, just the consumer and the dependency.